src/workerd/api/actor-state.c++

```diff
@@ -400,8 +397,7 @@ jsg::Promise<void> DurableObjectStorageOperations::setAlarm(kj::Date scheduledTi
   JSG_REQUIRE(scheduledTime > kj::origin<kj::Date>(), TypeError,
     "setAlarm() cannot be called with an alarm time <= 0");
 
-  JSG_REQUIRE(IoContext::current().getActorOrThrow().hasAlarmHandler(), TypeError,
-    "Your Durable Object class must have an alarm() handler in order to call setAlarm()");
+  IoContext::current().getActorOrThrow().assertCanSetAlarm();
```

Comment:
Non-blocking nit: It might be good here just to add a comment that `assertCanSetAlarm` can fall-through unchecked if called during init, just for awareness for someone new coming into this code.